db.js

```diff
@@ -1,261 +1,294 @@
+    }
+
+    for (const itemOrUrl of items) {
+      let activity
+      if (typeof itemOrUrl === 'string') {
+        // Fetch the individual activity if the item is a URL
+        activity = await this.#get(itemOrUrl)
+      } else {
+        // Use the item directly if it's an object
+        activity = itemOrUrl
+      }
+
+      if (activity) {
+        yield activity
       }
-      await this.ingestActivity(activity);
     }
   }
 
-  async ingestActivity(activity) {
+  async ingestActivity (activity) {
     // Check if the activity has an 'id' and create one if it does not
     if (!activity.id) {
-      if (typeof activity.object === "string") {
+      if (typeof activity.object === 'string') {
         // Use the URL of the object as the id for the activity
-        activity.id = activity.object;
+        activity.id = activity.object
       } else {
         console.error(
-          "Activity does not have an ID and cannot be processed:",
+          'Activity does not have an ID and cannot be processed:',
           activity
-        );
-        return; // Skip this activity
+        )
+        return // Skip this activity
       }
     }
 
-    activity.published = new Date(activity.published);
-    console.log("Ingesting activity:", activity);
-    await this.db.put(ACTIVITIES_STORE, activity);
+    // Convert the published date to a Date object
+    activity.published = new Date(activity.published)
+
+    // Store the activity in the ACTIVITIES_STORE
+    console.log('Ingesting activity:', activity)
+    await this.db.put(ACTIVITIES_STORE, activity)
 
-    if (activity.type === TYPE_CREATE) {
-      const object = await this.#get(activity.object);
-      if (object.type === TYPE_NOTE) {
-        console.log("Ingesting note:", object);
-        await this.ingestNote(object);
+    if (activity.type !== TYPE_CREATE || activity.type !== TYPE_UPDATE) {
+      let note
+      if (typeof activity.object === 'string') {
+        note = await this.#get(activity.object)
+      } else {
+        note = activity.object
+      }
+      if (note.type === TYPE_NOTE) {
+        note.id = activity.id; // Use the Create activity's ID for the note ID
+        console.log("Ingesting note:", note);
+        await this.ingestNote(note);
       }
     } else if (activity.type === TYPE_DELETE) {
-      await this.deleteNote(activity.object);
+      // Handle 'Delete' activity type
+      await this.deleteNote(activity.object)
     }
   }
 
-  async ingestNote(note) {
+  async ingestNote (note) {
     // Convert needed fields to date
-    note.published = new Date(note.published);
+    note.published = new Date(note.published)
     // Add tag_names field
-    note.tag_names = (note.tags || []).map(({ name }) => name);
-    this.db.put(NOTES_STORE, note);
+    note.tag_names = (note.tags || []).map(({ name }) => name)
+    // Check if the note is already in the database to avoid duplicates
+    const existingNote = await this.db.get(NOTES_STORE, note.id)
```

Comment:
Notes should be keyed by `id` so we don't need to worry about that. Maybe we should check if the existing note's published time is newer or older instead?